### ISSUE TITLE
Failure with imagesAsPng=false and vectors for xdata&ydata; fix included

### ISSUE THREAD

**Issue by andreas12345** [matlab2tikz/matlab2tikz#713]
The following code fails with current m2t-develop:
```matlab
figure; imagesc(1:4,1:8,zeros(8,4))
matlab2tikz('/tmp/test.tikz','imagesAsPng',false)
```
Error message
```
Error using matlab2tikz>imageAsTikZ (line 2165)
Array lengths not matching (8 = size(cdata,1) ~= length(xData) = 4).

Error in matlab2tikz>drawImage (line 2061)
        [m2t, str] = imageAsTikZ(m2t, handle, xData, yData, cData);

Error in matlab2tikz>handleAllChildren (line 641)
                [m2t, str] = drawImage(m2t, child);

Error in matlab2tikz>drawAxes (line 848)
    [m2t, childrenEnvs] = handleAllChildren(m2t, handle);

Error in matlab2tikz>saveToFile (line 442)
        m2t = drawAxes(m2t, relevantAxesHandle);

Error in matlab2tikz (line 338)
saveToFile(m2t, fid, fileWasOpen);
```

The reason is that x and y dimensions of the data matrix are swapped in m2t, which is easy to fix:

```patch
diff --git a/src/matlab2tikz.m b/src/matlab2tikz.m
index 4a1fda4..69ab48d 100644
--- a/src/matlab2tikz.m
+++ b/src/matlab2tikz.m
@@ -2159,7 +2159,7 @@ function [m2t, str] = imageAsTikZ(m2t, handle, xData, yData, cData)
     switch length(xData)
         case 2 % only the limits given; common for generic image plots
             hX = 1;
-        case size(cData,1) % specific x-data is given
+        case size(cData,2) % specific x-data is given
             hX = (xData(end)-xData(1)) / (length(xData)-1);
         otherwise
             error('drawImage:arrayLengthMismatch', ...
@@ -2170,7 +2170,7 @@ function [m2t, str] = imageAsTikZ(m2t, handle, xData, yData, cData)
     switch length(yData)
         case 2 % only the limits given; common for generic image plots
             hY = 1;
-        case size(cData,2) % specific y-data is given
+        case size(cData,1) % specific y-data is given
             hY = (yData(end)-yData(1)) / (length(yData)-1);
         otherwise
             error('drawImage:arrayLengthMismatch', ...
```

Apologies for not posting a proper pull request with test case.

**Comment by egeerardyn** [comment]
Thank you for this bug report and fix :smile: , I will incorporate a fix shortly.

Just out of curiosity: what's the reason you are using TikZ output for images and not PNG?

**Comment by andreas12345** [comment]
Thank you for comitting the fix.

The reason why I did not want to use PNG in this case is that I have a plot of a matrix which is only 10 pixels wide. If I use PNG for that, some PDF viewers will interpolate across these 10 pixels, and it will not make sense anymore. Since the TikZ output has other disadvantages, I now use another workaround (upsampling the matrix before using imagesc).

**Comment by egeerardyn** [comment]
That PNG interpolation can indeed be a nuisance. I am thinking whether we could replace our current code with something more elegant (which hopefully compiles more quickly as well).

Thank you for the fix and the comments :+1:

**Comment by PeterPablo** [comment]
Fixed by #714.
